src/stores/filingHistoryListStore.ts

```diff
@@ -156,19 +161,32 @@ export const useFilingHistoryListStore = defineStore('filingHistoryList', {
      * @param context the Vuex context (passed in automatically)
      * @param id a business id or temporary registration number
      */
-    loadFilings (id: string): Promise<any> {
-      // need to return a promise
-      return new Promise((resolve, reject) => {
-        LegalServices.fetchFilings(id)
-          .then(filings => {
-            this.setFilings(filings)
-            // return the filings list
-            resolve(filings)
-          })
-          .catch(error => {
-            reject(error)
-          })
-      })
+    async loadFilings (id: string): Promise<any> {
+      try {
+        const filings = await LegalServices.fetchFilings(id)
+        this.setFilings(filings)
+        return filings
+      } catch (error) {
+        console.log('loadFilings() error =', error)
+      }
+    },
```

Comment:
I think this didn't work when we first tried it, because it's an action. (That may have been a limitation in Vuex but we now use Pinia.) So it works? Nice cleanup, thanks!

Does this return a more specific promise now?

Reply:
I can update the return type

Reply:
LGTM

Reply:
I have confirmed that the new code works. Hooray! I hated the previous code.